modules/gapi/test/streaming/gapi_streaming_tests.cpp

```diff
@@ -0,0 +1,199 @@
+    cv::GComputation c(in, out);
+
+    // Input data
+    cv::Mat in_mat = cv::imread(findDataFile("cv/edgefilter/kodim23.png"));
+    cv::Mat out_mat_gapi;
+
+    // OpenCV reference image
+    cv::Mat out_mat_ocv;
+    {
+        cv::Mat tmp;
+        cv::resize(in_mat(crop_rc), tmp, resample_sz);
+        cv::Canny(tmp, out_mat_ocv, thr_lo, thr_hi);
+    }
+
+    // Compilation & testing
+    auto ccomp = c.compileStreaming(cv::descr_of(in_mat),
+                                    cv::compile_args(cv::gapi::use_only{GetParam()}));
+    EXPECT_TRUE(ccomp);
+    EXPECT_FALSE(ccomp.running());
+
+    ccomp.setSource(cv::gin(in_mat));
+
+    ccomp.start();
+    EXPECT_TRUE(ccomp.running());
+
+    // Fetch the result 15 times
+    for (int i = 0; i < 15; i++) {
+        // With constant inputs, the stream is endless so
+        // the blocking pull() should never return `false`.
+        EXPECT_TRUE(ccomp.pull(cv::gout(out_mat_gapi)));
+        EXPECT_EQ(0, cv::countNonZero(out_mat_gapi != out_mat_ocv));
+        if (cv::countNonZero(out_mat_gapi != out_mat_ocv) > 0) {
+            cv::imshow("G-API", out_mat_gapi);
+            cv::imshow("OpenCV", out_mat_ocv);
+            cv::imshow("Delta", out_mat_gapi - out_mat_ocv);
+            cv::waitKey(0);
+        }
+    }
+
+    EXPECT_TRUE(ccomp.running());
+    ccomp.stop();
+
+    EXPECT_FALSE(ccomp.running());
+}
+
+TEST_P(GAPI_Streaming, SmokeTest_VideoInput_GMat)
+{
+    const auto crop_rc = cv::Rect(13, 75, 377, 269);
+    const auto resample_sz = cv::Size(224, 224);
+    const auto thr_lo = 64.;
+    const auto thr_hi = 192.;
+
+    cv::GMat in;
+    auto roi = cv::gapi::crop(in, crop_rc);
+    auto res = cv::gapi::resize(roi, resample_sz);
+    auto out = cv::gapi::Canny(res, thr_lo, thr_hi);
+    cv::GComputation c(cv::GIn(in), cv::GOut(cv::gapi::copy(in), out));
+
+    // OpenCV reference image code
+    auto opencv_ref = [&](const cv::Mat &in_mat, cv::Mat &out_mat) {
+        cv::Mat tmp;
+        cv::resize(in_mat(crop_rc), tmp, resample_sz);
+        cv::Canny(tmp, out_mat, thr_lo, thr_hi);
+    };
+
+    // Compilation & testing
+    auto ccomp = c.compileStreaming(cv::GMatDesc{CV_8U,3,cv::Size{768,576}},
+                                    cv::compile_args(cv::gapi::use_only{GetParam()}));
+    EXPECT_TRUE(ccomp);
+    EXPECT_FALSE(ccomp.running());
+
+    ccomp.setSource(cv::gapi::GVideoCapture{findDataFile("cv/video/768x576.avi")});
+
+    ccomp.start();
+    EXPECT_TRUE(ccomp.running());
+
+    // Process the full video
+    cv::Mat in_mat_gapi, out_mat_gapi;
+
+    while (ccomp.pull(cv::gout(in_mat_gapi, out_mat_gapi))) {
+        cv::Mat out_mat_ocv;
+        opencv_ref(in_mat_gapi, out_mat_ocv);
+        EXPECT_EQ(0, cv::countNonZero(out_mat_gapi != out_mat_ocv));
+    }
+    EXPECT_FALSE(ccomp.running());
+
+    // Stop can be called at any time (even if the pipeline is not running)
+    ccomp.stop();
+
+    EXPECT_FALSE(ccomp.running());
+}
+
```

Comment:
...and unfortunately there's no straight way to make it at the moment of `setSource()`  (since the capture starts reading the stream only after `start()`). Passing exceptions through the pipeline is also cumbersome and error-prone. Better to remove this completely and force the "reshape()" functionality.